### PR TITLE
Comment for testing build

### PR DESCRIPTION
**Pull request by TomerAdmon**
now we can trigger build with those properties:
- personal build
- comment
- queue on top
- clean sources
- require rebuild dependencies

acceptance tests were fixed.

### CODE REVIEW

File: FluentTc/Locators/MoreOptionsHavingBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    public interface IMoreOptionsHavingBuilder
+    {
+        IMoreOptionsHavingBuilder WithCleanSources();
+        IMoreOptionsHavingBuilder RebuildAllDependencies();
+        IMoreOptionsHavingBuilder QueueAtTop();
+        IMoreOptionsHavingBuilder AsPersonal();
+        IMoreOptionsHavingBuilder WithComment(string comment);
+    }
+
+    public class MoreOptionsHavingBuilder : IMoreOptionsHavingBuilder
```
Comment (borismod) [inline FluentTc/Locators/MoreOptionsHavingBuilder.cs]: MoreOptionsHavingBuilder should not be visible by user of FluentTc. Usually such classes are internal

File: FluentTc/Engine/BuildConfigurationRunner.cs [hunk 2]
```diff
@@ -49,17 +63,48 @@ private static List<Property> GetProperties(Action<IBuildParameterValueBuilder>
...
 
             if (agentId.HasValue)
             {
                 bodyBuilder.AppendFormat(@"<agent id=""{0}""/>", agentId).AppendLine();
             }
 
+            if (moreOptions != null &&
+                moreOptions.GetComment() != null)
+            {
+                bodyBuilder.AppendFormat(@"<comment><text>{0}</text></comment>", moreOptions.GetComment()).AppendLine();
```
Comment (borismod) [inline FluentTc/Engine/BuildConfigurationRunner.cs]: What happens if comment contains some characters that might break the XML, like < > /?

File: FluentTc/Engine/BuildConfigurationRunner.cs [hunk 3]
```diff
@@ -49,17 +63,48 @@ private static List<Property> GetProperties(Action<IBuildParameterValueBuilder>
...
+            }
+
+            if (moreOptions != null &&
+                (moreOptions.TriggeringOptions.CleanSources == true ||
```
Comment (borismod) [inline FluentTc/Engine/BuildConfigurationRunner.cs]: There is no need to check booleans with "==true", it's enough to check:
```C# 
if ( moreOptions.TriggeringOptions.CleanSources )
{
}
```

Comment (borismod) [inline FluentTc/Engine/BuildConfigurationRunner.cs]: Don;t you need also to check 
```C#
if ( moreOptions.TriggeringOptions.HasValue )
```

**Comment by borismod** [comment]
@TomerAdmon, great contribution :+1: 
I wrote a few minor notes, could you take a look :question:

**Comment by TomerAdmon** [comment]
Hi, 
Thanks for the review.
1) fixed
2) fixed + add a test
3) regarding the ==true, since this is a nullable bool, we need to use the ==True (see [here](http://stackoverflow.com/questions/2673918/best-way-to-check-for-nullable-bool-in-a-condition-expression-if))
4) regarding the "if ( moreOptions.TriggeringOptions.HasValue )", we check that the moreOptions is not null, and if it's not null it means we initialized the ctor that created a new TriggeringOptions... so no need to test it....

File: FluentTc/Locators/MoreOptionsHavingBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            return m_Comment;
+        }
+
+        public TriggeringOptions TriggeringOptions
+        {
+            get { return m_TriggeringOptions; }
+        }
+    }
+
+    public class TriggeringOptions
```
Comment (borismod) [inline FluentTc/Locators/MoreOptionsHavingBuilder.cs]: Can be also internal, isn't it?

**Pull request merged**